module5/src/com/epam/calendar/entity/Calendar.java

```diff
@@ -0,0 +1,164 @@
+package com.epam.calendar.entity;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+public class Calendar {
+    private int year;
+    private List<Holiday> holidays;
+    private List<Weekend> weekends;
+    private static byte NUMBER_OF_FIRST_MONTH = 1;
+    private static byte NUMBER_OF_FIRST_DAY = 1;
+
+    public Calendar(int year) {
+        this.year = year;
+        this.holidays = new ArrayList<>();
+        this.weekends = new ArrayList<>();
+        addWeekend(year);
+    }
+
+    private void addWeekend(int year) {
+        LocalDate date = LocalDate.of(year, NUMBER_OF_FIRST_MONTH,NUMBER_OF_FIRST_DAY);
+        while (date.getDayOfWeek() != DayOfWeek.SATURDAY && date.getDayOfWeek() != DayOfWeek.SUNDAY) {
+            date = date.plusDays(1);
+        }
+        while (date.getYear() == year) {
+            if (date.getDayOfWeek() == DayOfWeek.SATURDAY) {
+                weekends.add(new Weekend(date, "Saturday"));
+                date = date.plusDays(1);
+            } else if (date.getDayOfWeek() == DayOfWeek.SUNDAY) {
+                weekends.add(new Weekend(date, "Sunday"));
+                date = date.plusDays(1);
+            } else {
+                date = date.plusDays(1);
+            }
+        }
```

Comment:
I think this is crazy to check every day when you know that week has 7 days and 2 weekends in it...